Sources/Core/API/TextureManager.swift

```diff
@@ -93,13 +99,24 @@ public final class TextureManager {
                 return nil
             }
 
-            return load(texture, namePrefix: namePrefix, scale: scale - 1)
+            return load(texture, namePrefix: namePrefix,
+                        scale: otherAllowedScales.removeLast(), otherAllowedScales: otherAllowedScales)
         }
 
         let texture = LoadedTexture(image: image, scale: scale)
         cache[cacheKey] = texture
         return texture
     }
+
+    // MARK: - Private
+
+    func allowedScales(beside defaultValue: Int?) -> [Int] {
+        let minScale = 1
+        let maxScale = maxAllowedScale
+
+        return [Int](minScale...maxScale)
```

Comment:
Since this code is in a very performance-ciritical path (loading textures, which can occur many times per frame), I don't think we should create multiple collections like this. Instead, how about simply iterating by incrementing the scale up to the maximum point, just like we do when scaling downwards?

Reply:
@JohnSundell, I'm sorry, I somehow missed your report 😏 

So you're saying I should just iterate up, and then down to 1? 
I agree that performance may and probably will be the issue here. I'll try to 

As for the tests, it's like I've stated in the opening of the pull request. They got broken because they've been searching for images that did not exist, so multiple resolutions added to names collection, without any images corresponding to them.

As you say, I also think that new code should not break the tests, but this one broke. Also, I've forgotten to actually add test for the new scaling :/

Reply:
I just have this suspicion that the tests worked because the scaling went downwards.

Reply:
What I mean is, when the test looked like this:
```swift
func testAnimatingWithSpriteSheet() {
        let imageSize = Size(width: 300, height: 100)
        let image = ImageMockFactory.makeImage(withSize: imageSize)
        game.textureImageLoader.images["sheet.png"] = image.cgImage

        var animation = Animation(
            spriteSheetNamed: "sheet",
            frameCount: 6,
            rowCount: 2,
            frameDuration: 1
        )
        animation.textureScale = 1

        let actor = Actor()
        actor.animation = animation
        game.scene.add(actor)
        XCTAssertEqual(actor.size, Size(width: 100, height: 50))

        game.timeTraveler.travel(by: 1)
        game.update()
        XCTAssertEqual(actor.size, Size(width: 100, height: 50))

//        game.textureImageLoader.images["sheet2.png"] = image.cgImage

        // Assigning new sprite sheet mid-animation should update the animation
        var newAnimation = Animation(
            spriteSheetNamed: "sheet2",
            frameCount: 6,
            rowCount: 2,
            frameDuration: 1
        )
        newAnimation.textureScale = 1
        actor.animation = newAnimation

        game.update()

        XCTAssertEqual(game.textureImageLoader.imageNames, ["sheet.png", "sheet2.png"])
    }
```

I got `XCTAssertEqual failed: ("["sheet.png", "sheet2@3x.png", "sheet2.png", "sheet2@2x.png"]") is not equal to ("["sheet.png", "sheet2.png"]")`

Reply:
So I have probably broken something